src/OpenSage.Game/Logic/Object/Draw/W3dModelDraw.cs

```diff
@@ -119,6 +119,20 @@ public override void UpdateConditionState(BitArray<ModelConditionFlag> flags)
             }
 
             SetActiveConditionState(bestConditionState);
+
+            bestConditionState.WeaponMuzzleFlashes.ForEach(x => {
```

Comment:
Nitpick - please avoid lambdas / delegates in anything that happens in the main game loop. Lambdas cause memory allocation, and therefore garbage collection. This can be rewritten as a standard `foreach`.

Reply:
Rewrote it using a foreach